localnet/poktrolld/config/relayminer_config_full_example.yaml

```diff
@@ -0,0 +1,108 @@
+    # Required.
+    type: http
+    # Hostname to open port on. Use 0.0.0.0 in containerized environments,
+    # 127.0.0.1 with a reverse-proxy when there's another process on localhost
+    # that can be used as a reverse proxy (nginx, apache, traefik, etc.).
+    # Required.
+    host: 127.0.0.1:8080
+
+  # TODO_IMPROVE: https is not currently supported, but this is how it could potentially look.
+  # - name: example-how-we-can-support-https
+  #   type: https
+  #   host: 0.0.0.0:8443
+  #   tls:
+  #     enabled: true
+  #     certificate: /path/to/crt
+  #     key: /path/to/key
+
+# Suppliers are different services that can be offered through RelayMiner.
+# When a supplier is configured to use a proxy and staked appropriately,
+# the relays will start flowing through RelayMiner.
+suppliers:
+    # Name of the supplier offered to the network.
+    # Must be unique.
+    # Required.
+  - name: ethereum
+    # Type of how the supplier offers service through the network.
+    # Must match the type of the proxy the supplier is connected to.
+    # Required.
+    type: http
+    # Configuration of the service offered through RelayMiner.
+    service_config:
+      # URL RelayMiner proxies the requests to.
+      # Required.
+      url: http://anvil.servicer:8545
+      # Authentication for the service.
+      # HTTP Basic Auth: https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication
+      # Optional.
+      authentication:
+        username: user
+        password: pwd
+
+      # TODO_IMPROVE: This is not supported in code yet,
+      # but some services authenticate via a header.
+      # Example, if the service requires a header like `Authorization: Bearer <PASSWORD>`
+      # Authorization: Bearer <PASSWORD>
+      # Optional.
+      headers: {}
+
+    # A list of hosts the HTTP service is offering.
+    # When linked to the proxy, that hostname is going to be used to route the
+    # request to the correct supplier.
+    # That hostname is what the user should stake the supplier for.
+    # Must be unique within a proxy/proxies it is set up on;
+    # in other words, one proxy can't offer the same hostname more than once.
+    # The `name` of the supplier is automatically added to the hosts section
+    # for potential troubleshooting/debugging purposes
+    # Required.
+    hosts:
+      - ethereum.devnet1.poktroll.com
+      # - ethereum # <- this part is be added automatically.
+
+    # Names of proxies that this supplier is connected to.
+    # This MUST correspond to the name in the `proxies` section
+    # in order for the supplier to be accessible to the external network.
+    # Required.
+    proxy_names:
+      - http-example # when the RelayMiner server builder runs.
+  - name: 7b-llm-model
+    type: http
+    service_config:
+      url: http://llama-endpoint
+    hosts:
+      - 7b-llm-model.devnet1.poktroll.com
+      # - 7b-llm-model # <- this part can be added automatically.
```

Comment:
can or will?

I'm still a bit confused why hosts is a list so #PUC

Reply:
This **is** added automatically.

@okdas  may provide more arguments.

The main reasons I see are
* That the operator may want to route the requests coming from these domains to a single proxy
* Migrating to a new domain, where requests may still arrive from the old one 
* Having internal requests coming from non FQDNs because of complex routing and debugging purposes.

Reply:
@red-0ne is correct. If we can support multiple, I'd say this is the way. It allows for easier migrations across domains, it gives more flexibility. Kubernetes Ingresses work the same way, heres what ChatGPT has to say about why that is: https://chat.openai.com/share/286502d2-021b-4caa-8c90-eb6138a57c01 - not all points apply to our case (yet)

Reply:
@red-0ne I know you already documneted a subset of these, but I think there's opportunity to add even a few more of the points here. Make sure to add the appropriate qualifiers (see my comment)

Reply:
Updated on the `ethereum` example entry